services/datamanager/builtin/builtin.go

```diff
@@ -483,8 +503,15 @@ func (svc *builtIn) Update(ctx context.Context, cfg *config.Config) error {
 	newCollectorMetadata := make(map[componentMethodMetadata]bool)
 	for _, attributes := range allComponentAttributes {
 		if !attributes.Disabled && attributes.CaptureFrequencyHz > 0 {
-			componentMetadata, err := svc.initializeOrUpdateCollector(
+
+			componentMetadata, captureConfigsChanged, err := svc.initializeOrUpdateCollector(
 				attributes, updateCaptureDir)
+
+			if !sessionIDUpdated && captureConfigsChanged {
```

Comment:
The session ID gets updated after the collectors are initialized/updated in `initializeOrUpdateCollector`. This session ID is correctly new if `updateCaptureDir || toggledCaptureOn` but not if the collector configuration has changed. Adding my suggestion for a fix in another comment in that method.